shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -51,6 +51,143 @@
+                  methodCall:@[ kExtendSelectionToNextWordBoundaryIntent, @(true), @(false) ]],
+  [SelectorIntent intentWith:@selector(moveToBeginningOfLineAndModifySelection:)
+                  methodCall:@[ kExtendSelectionToLineBreakIntent, @(false), @(false) ]],
+  [SelectorIntent intentWith:@selector(moveToEndOfLineAndModifySelection:)
+                  methodCall:@[ kExtendSelectionToLineBreakIntent, @(true), @(false) ]],
+  [SelectorIntent intentWith:@selector(moveToBeginningOfDocumentAndModifySelection:)
+                  methodCall:@[ kExtendSelectionToDocumentBoundaryIntent, @(false), @(false) ]],
+  [SelectorIntent intentWith:@selector(moveToEndOfDocumentAndModifySelection:)
+                  methodCall:@[ kExtendSelectionToDocumentBoundaryIntent, @(true), @(false) ]],
+
+  // TODO: https://github.com/flutter/flutter/issues/78660
+  // These intents currently move the caret based on the logical
+  // order of surrounding characters instead of their visual order.
+  [SelectorIntent intentWith:@selector(moveLeft:)
+                  methodCall:@[ kExtendSelectionByCharacterIntent, @(false), @(true) ]],
+  [SelectorIntent intentWith:@selector(moveRight:)
+                  methodCall:@[ kExtendSelectionByCharacterIntent, @(true), @(true) ]],
+  [SelectorIntent intentWith:@selector(moveWordLeft:)
+                  methodCall:@[ kExtendSelectionToNextWordBoundaryIntent, @(false), @(true) ]],
+  [SelectorIntent intentWith:@selector(moveWordRight:)
+                  methodCall:@[ kExtendSelectionToNextWordBoundaryIntent, @(true), @(true) ]],
+  [SelectorIntent intentWith:@selector(moveToLeftEndOfLine:)
+                  methodCall:@[ kExtendSelectionToLineBreakIntent, @(false), @(true) ]],
+  [SelectorIntent intentWith:@selector(moveToRightEndOfLine:)
+                  methodCall:@[ kExtendSelectionToLineBreakIntent, @(true), @(true) ]],
+
+  [SelectorIntent intentWith:@selector(moveLeftAndModifySelection:)
+                  methodCall:@[ kExtendSelectionByCharacterIntent, @(false), @(false) ]],
+  [SelectorIntent intentWith:@selector(moveRightAndModifySelection:)
+                  methodCall:@[ kExtendSelectionByCharacterIntent, @(true), @(false) ]],
+  [SelectorIntent intentWith:@selector(moveWordLeftAndModifySelection:)
+                  methodCall:@[ kExtendSelectionToNextWordBoundaryIntent, @(false), @(false) ]],
+  [SelectorIntent intentWith:@selector(moveWordRightAndModifySelection:)
+                  methodCall:@[ kExtendSelectionToNextWordBoundaryIntent, @(true), @(false) ]],
+  [SelectorIntent intentWith:@selector(moveToLeftEndOfLineAndModifySelection:)
+                  methodCall:@[ kExtendSelectionToLineBreakIntent, @(false), @(false) ]],
+  [SelectorIntent intentWith:@selector(moveToRightEndOfLineAndModifySelection:)
+                  methodCall:@[ kExtendSelectionToLineBreakIntent, @(true), @(false) ]]
+
+  // TODO: implement paragrah boundary intents in the framework.
```

Comment:
Heads up that this TODO is still here.

Reply:
We don't have paragraph boundary defined in the framework right now. I'll create an issue for that.